staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -720,6 +729,102 @@ func ValidateCustomResourceColumnDefinition(col *apiextensions.CustomResourceCol
 	return allErrs
 }
 
+func validateFieldPath(fieldPath string, pathOfFieldPath *field.Path, schema *apiextensions.JSONSchemaProps) field.ErrorList {
```

Comment:
While not urgent for this enhancement, this is a really powerful function and might be easier to leverage in future work if organized in a small library for JSON Path processing. I'm okay with this happening post-freeze but wanted to mention the idea and encourage the parsing code to be organized so this would be possible to do in the future. For example, having unit tests specifically for this function seem valuable.
To that end, godoc seems worthwhile on this function even though it is currently unexported.

Reply:
I have added unit tests for this func.